undertow/src/main/java/org/wildfly/extension/undertow/UndertowSubsystemAdd.java

```diff
@@ -102,8 +105,9 @@ protected void performBoottime(OperationContext context, ModelNode operation, Re
         final String defaultSecurityDomain = UndertowRootDefinition.DEFAULT_SECURITY_DOMAIN.resolveModelAttribute(context, model).asString();
 
         final ModelNode instanceIdModel = UndertowRootDefinition.INSTANCE_ID.resolveModelAttribute(context, model);
-        final String instanceId = instanceIdModel.isDefined() ? instanceIdModel.asString() : null;
-
+        final ModelNode uuidInstanceIdModel = UndertowRootDefinition.UUID_INSTANCE_ID.resolveModelAttribute(context, model);
+        final String instanceId = instanceIdModel.isDefined() ? (uuidInstanceIdModel.asBoolean() ?
+                new String(Base64.getUrlEncoder().encode(instanceIdModel.asBytes())) : instanceIdModel.asString()) : null;
```

Comment:
This just generates a base64 encoded string from the `instanceIdModel` value. A base64 encoded string can be easily decoded to the original value (= hostname by default). For example, the original value can decode it by the following command:

```
$ echo -n <generate-encoded-value> | base64 -d
```

So, as this approach can not hide the actual hostname information, I think this is not what the customer wants. 

I would suggest generating a base64 encoded string from a MD5 hash digest of the `instanceIdModel` value. Java's name-based UUID (`UUID.nameUUIDFromBytes()`) is internally generated from MD5 hash digest of the input value. So, I think this approach is the same level of uniqueness to `UUID.nameUUIDFromBytes()`.

Also, `Base64.Encoder` generates an encoded string with a padding (`=` by default). However, `=` is not allowed in V0 Cookie. So, I think it's better to use `Base64.getUrlEncoder().withoutPadding()` instead.

For example:

```
        final String instanceId = instanceIdModel.isDefined() ?
            (uuidInstanceIdModel.asBoolean() ? generateObfuscatedInstanceId(instanceIdModel.asString()) : instanceIdModel.asString()): null;

    ...(snip)...

    private String generateObfuscatedInstanceId(String instanceId) {
        try {
            MessageDigest md = MessageDigest.getInstance("MD5");
            byte[] md5Bytes = md.digest(instanceId.getBytes());
            Base64.Encoder encoder = Base64.getUrlEncoder().withoutPadding(); // = is not allowed in V0 Cookie
            String encodedString = encoder.encodeToString(md5Bytes);
            UndertowLogger.ROOT_LOGGER.infof("Obfuscated value [%s] is generated from [%s] and will be used as instance-id (jvmRoute)", encodedString, instanceId);
            return encodedString;
        } catch (NoSuchAlgorithmException e) {
            throw new IllegalArgumentException(e);
        }
    }
```

Reply:
This may take the problem full circle but are they concerned about these values being mapped to known host names?  If so maybe some additional secret data is needed in the hash to prevent comparison against known host names.

Reply:
@darranl  thanks for your feedback. I'll ask if that's a concern or not. is that something complex to do, or something super simple? The question here is, if the additional secret data is interesting to the user, should it be a follow up RFE or should we add it to this RFE? Please let me know what you think about this.

Reply:
@darranl is referring to a salt.  I think it is sufficient to use the Undertow server name as the salt, as the route identifier (which is the value we are actually interested in) is generated per Undertow server.  This is sufficient to prevent comparison against known host names.

Reply:
+1 provided that server name is not shared publicly it should be sufficient in this case.

Where only public information goes into a hash it is very easy for someone to create their own reference tables in a kind of reverse lookup.  Once some non public information is added it becomes a lot more complex.

The piece I am not familiar with here is the full use of these fields in case my suggestion made it much more complex where we need to be able to map back.

Reply:
The name of an Undertow server resource is effectively arbitrary, and has no intrinsic meaning outside of the subsystem model - hence my suggestion to use it as a salt (since the name of an Undertow server resource is unique per server).

Both the instance-id and the names of Undertow servers stored as clear text in the model, so there should be no need/reason to map back.

Reply:
@pferraro that does sound like enough then.

If anyone wanted more than that we would likely need to be looking at stronger approaches, possibly with keys etc...